modules/monitoring-setting-the-body-size-limit-for-metrics-scraping.adoc

```diff
@@ -0,0 +1,61 @@
+// Module included in the following assemblies:
+//
+// * monitoring/configuring-the-monitoring-stack.adoc
+
+:_content-type: PROCEDURE
+[id="setting-the-body-size-limit-for-metrics-scraping_{context}"]
+= Setting the body size limit for metrics scraping
+
+By default, no limit is enforced on the uncompressed body size of scraped metrics targets.
+You can set a body size limit to help avoid situations in which Prometheus consumes excessive amounts of memory when scraped targets return a response that contains a large amount of data.
+In addition, by setting a body size limit, you can reduce the impact that a malicious target could have on Prometheus and on the cluster as a whole.
+
+After you set a value for `enforcedBodySizeLimit`, the alert `PrometheusScrapeBodySizeLimitHit` fires when at least one Prometheus scrape target replies with a response body larger than the configured value.
+
+[NOTE]
+====
+Metrics scraped from targets that respond with an uncompressed body size exceeding the configured size limit cause the scrape to fail.
+The targets are considered to be down and have their `up` metric set to `0`, which could trigger the `TargetDown` alert.
+====
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have installed the OpenShift CLI (`oc`).
+
+.Procedure
+
+. Edit the `cluster-monitoring-config` `ConfigMap` object in the `openshift-monitoring` namespace:
++
+[source,terminal]
+----
+$ oc -n openshift-monitoring edit configmap cluster-monitoring-config
+----
+
+. Add a value for `enforcedBodySizeLimit` to `data/config.yaml/prometheusK8s` to limit the body size that can be accepted per target scrape:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |-
+    prometheusK8s:
+      enforcedBodySizeLimit: 40MB <1>
+----
+<1> Specify the maximum body size for scraped metrics targets.
+This `enforcedBodySizeLimit` example limits the uncompressed size per target scrape to 40 megabytes.
```

Comment:
Shall we mention the supported units: B, KB, MB, GB, TB, PB, EB ?

Reply:
Added `B (bytes), KB (kilobytes), MB (megabytes), GB (gigabytes), TB (terabytes), PB (petabytes), and EB (exabytes)` to the options.